src/hooks/useIsOnScreen.ts

```diff
@@ -0,0 +1,41 @@
+import { RefObject, useEffect, useRef, useState } from 'react';
+
+export const useIsOnScreen = (ref: RefObject<HTMLElement>) => {
+  const observerRef = useRef<IntersectionObserver | null>(null);
+  const [isOnScreen, setIsOnScreen] = useState(true);
+
+  useEffect(() => {
+    if (ref.current) {
+      const scrollParent = getScrollParent(ref.current);
+      observerRef.current = new IntersectionObserver(
+        ([entry]) => {
+          setIsOnScreen(entry.isIntersecting);
+        },
+        {
+          root: scrollParent === window ? undefined : (scrollParent as HTMLElement),
+          rootMargin: '200px 0px 0px 0px',
```

Comment:
shouldn't we use a viewport unit? Example:
```suggestion
          rootMargin: '20vh 0px 0px 0px',
```

Reply:
<img width="813" alt="Screen Shot 2021-12-01 at 4 33 45 PM" src="https://user-images.githubusercontent.com/847542/144317543-57150437-b817-4ed6-8224-3f761760eb7e.png">

Reply:
Meant to lower to 100px though, doing that.

Reply:
yeah, percentages feel a bit weird. 100px works.